### PR TITLE
Raise Dead (Random Event)

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Adds a new event, Raise Dead. When this occurs, random dead bodies will come back to life, given that they have no mind, they have a brain, a heart and have not sustained a lot of damage. 

These bodies are not antagonists, and they do not remember their past lives. They're not romerol zombies either and follow lavaland escalation.

## Why It's Good For The Game

Adds new ways for dead people to play the game. Allows for roleplaying options. 

Dead bodies are technically valid, like golems, meaning chaplain can hunt then for whatever reason, and people can enslave them and abuse them. 

## Progress
* [x] Base Event
* [x] Body Check
* [x] Revitalize Mobs
* [x] Testing

## Changelog
:cl:
add: Raise Dead, a new random event
/:cl:

### CODE REVIEW

File: code/modules/events/undead.dm [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+/datum/round_event_control/undead
+	name = "Raise Dead"
+	typepath = /datum/round_event/ghost_role/undead
+	weight = 10
+
+/datum/round_event/ghost_role/undead
+	minimum_required = 1
+	role_name = "friendly zombie"
+	var/spawns = 3
+	var/one = "one"
```
Comment (Archanial) [inline code/modules/events/undead.dm]: ???

File: code/modules/events/undead.dm [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	fakeable = TRUE
+
+/datum/round_event/ghost_role/undead/announce(fake)
+	var/source = pick("a powerful spell from a far away wizard", "a strange anomalous enegy", "the influence of a high level necromancer", "the power of love", "the will of an omnipotent God", "\[REDACTED\]")
+	var/purpose = pick("party like there's no tomorrow", "seek employment", "partake in the democratic process", "fulfill their new year's promises", "have an after-life crysis", "preach about their new religion", "answer your questions about life, the universe, and everything", "persuade you not to treat them like their brain eating counterparts", "repair their past mistakes", "\[REDACTED\]")
+
+	priority_announce("Your station has been subjected to [source]. Some dead bodies have come back to life to [purpose]. On behalf of Nanotrasen, please welcome these former employees as your new coworkers!","[command_name()] Medium-Priority Update")
+
+/datum/round_event/ghost_role/undead/spawn_role()
+	var/list/mob/dead/observer/candidates = get_candidates(ROLE_REVENANT, null, ROLE_REVENANT)
```
Comment (Archanial) [inline code/modules/events/undead.dm]: Role revenant?

**Comment by Effemu** [comment]
"lavaland escalation"

>shitsec fires disabler
>shove sec
>you are now **valid™️**

File: code/modules/events/undead.dm (hunk 1 above)
Comment (eeSPee) [reply code/modules/events/undead.dm]: Oops leftover.

File: code/modules/events/undead.dm (hunk 2 above)
Comment (eeSPee) [reply code/modules/events/undead.dm]: Figured they're Undead, thus Revenant role.

**Comment by eeSPee** [comment]
> "lavaland escalation"
> 
> > shitsec fires disabler
> > shove sec
> > you are now **valid™️**

T'is the life of golem.

File: code/modules/events/undead.dm (hunk 2 above)
Comment (PowerfulBacon) [reply code/modules/events/undead.dm]: ?

Comment (eeSPee) [reply code/modules/events/undead.dm]: That line is from sentience potion, who uses the xenomorph role for some reason.

I think roles need a revision.

Comment (DatBoiTim) [reply code/modules/events/undead.dm]: To be fair current revenant is closer to a poltergheist than revenants.

File: code/modules/events/undead.dm [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+		zombie.set_species(/datum/species/zombie)
+
+		zombie.revive(full_heal = TRUE, admin_revive = FALSE)
+		zombie.regenerate_organs()
+
+		LAZYADD(spawned_mobs, zombie)
+
+		//Flavortext
+		to_chat(zombie, "<span class='userdanger'>Welcome back!</span>")
+		to_chat(zombie, "<span class='warning'>An unknown source of energy, be it magic or science, has put you back into your body! You have no memory of your past life, but others may have memory of you! You are not an antagonist, and shouldn't act as such.</span>")
```
Comment (utilbird) [inline code/modules/events/undead.dm:60]: Add this to their memory too

**Comment by TeomanTheGreat** [comment]
What if they are the target of an antagonist? How will the antagonist know that their target just came back to life. You could make it so it doesn't choose antag targets for the event.

**Comment by eeSPee** [comment]
> What if they are the target of an antagonist? How will the antagonist know that their target just came back to life. You could make it so it doesn't choose antag targets for the event.

Well it assigns a new mind to catatonic bodies. Objectives check for mind, not the target body. So it shouldn't be an issue.

**Comment by eeSPee** [comment]
Could be wizard event?

File: code/modules/events/undead.dm [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	fakeable = TRUE
+
+/datum/round_event/ghost_role/undead/announce(fake)
+	var/source = pick("a powerful spell from a far away wizard", "a strange anomalous enegy", "the influence of a high level necromancer", "the power of love", "the will of an omnipotent God", "\[REDACTED\]")
+	var/purpose = pick("party like there's no tomorrow", "seek employment", "partake in the democratic process", "fulfill their new year's promises", "have an after-life crysis", "preach about their new religion", "answer your questions about life, the universe, and everything", "persuade you not to treat them like their brain eating counterparts", "repair their past mistakes", "\[REDACTED\]")
+
+	priority_announce("Your station has been subjected to [source]. Some dead bodies have come back to life to [purpose]. On behalf of Nanotrasen, please welcome these former employees as your new coworkers!","[command_name()] Medium-Priority Update")
+
+/datum/round_event/ghost_role/undead/spawn_role()
+	var/list/mob/dead/observer/candidates = get_candidates(ROLE_LAVALAND, null, ROLE_LAVALAND)
```
Comment (PowerfulBacon) [inline code/modules/events/undead.dm:20]: Make a new role for this, the candidates is used for the No for the rest of the round option so should be unique for spawns.

**Pull request closed without merging**